frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -141,7 +143,7 @@ export type PrometheusResponse = {
 };
 
 export type WatchK8sResource = {
-  kind: K8sResourceKindReference;
+  kind: K8sResourceKindReference | K8sGroupVersionKind; // can use reference (old format) or K8sGroupVersionKind
```

Comment:
I think we should be more explicit. If anyone reads the type, I'd like it to encourage them to use the new type if we are trying to get away from the old one.

Something like this maybe?
```
/**
 * @deprecated - We are migrating towards using a group/version/kind object instead of the '~' concat string.
 */
type DeprecatedK8sReference = K8sResourceKindReference;

export type WatchK8sResource = {
  kind: K8sGroupVersionKind | DeprecatedK8sReference;
...
```

Reply:
Upon further thinking as I reviewed the rest. I think this is good, but I wonder we should not just deprecate the reference in `frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts` (and `GroupVersionKind` as well). Indicate that these are on the way out and to migrate any usage over to `K8sGroupVersionKind`

Reply:
I agree, deprecating `GroupVersionKind` type ASAP makes sense

Reply:
added `deprecated` for kind and introduced new prop  for type `K8sGroupVersionKind`